added "title" columns

alexrybch opened the pull request:
Signed-off-by: Alex Rybchenko <arybchenko@box.com>

## Hey, I just made a Pull Request!

Added functions to render recently added "title" field from entity metadata in EntityTable and CatalogTable components.
![image](https://user-images.githubusercontent.com/3530856/135136608-392fc573-e3b8-4b5c-9e6e-2bb5308619df.png)

Closes #7361 

#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

Rugvip reviewed (commented):
Makes sense, thanks! 👍 

Just need to add some changesets!

alexrybch commented:
added changeset :)

freben commented:
@Rugvip An alternative would be to have one column, that shows either

- the name, if that's all we have, like before
- the title if it exists, with the name as a tooltip or so (so you can kind of get at both)

There's a larger picture to this of course. For example in a page header, would we do the exact same? Or perhaps the title as the main title and the name as the subtitle? It's of course important to be able to conveniently be able to see both somehow

alexrybch commented:
@freben I like your idea with one column, will do the update!

dhenneke commented:
> @Rugvip An alternative would be to have one column, that shows either
> 
> * the name, if that's all we have, like before
> * the title if it exists, with the name as a tooltip or so (so you can kind of get at both)

It would be nice if the `EntityRefLink` component could magically display the title if available. This way it would be consistent everywhere. But sadly, the `EntityRefLink` only has the entity ref and would require a rest call to get the title information...

alexrybch commented:
I removed new column creators and updated `EntityRefLink` to accept `title' prop to override default entity title (default name is shown on hover). Also updated existing `createNameColumn` functions to pass `metadata.title' as title prop to `EntityRefLink`.
![image](https://user-images.githubusercontent.com/3530856/135279917-3a8e4df3-9345-47e4-bc78-a6ba8b4edf37.png)

Rugvip reviewed (commented):
Looks good again, and still just needs a changeset x)

For fixing the tests you can use `renderInTestApp` from `@backstage/test-utils` and pass in this as an option:

```ts
mountedRoutes: {
  '/catalog/:namespace/:kind/:name/*': entityRouteRef,
},
```

alexrybch commented:
> Looks good again, and still just needs a changeset x)
> 
> For fixing the tests you can use `renderInTestApp` from `@backstage/test-utils` and pass in this as an option:
> 
> ```ts
> mountedRoutes: {
>   '/catalog/:namespace/:kind/:name/*': entityRouteRef,
> },
> ```

my bad, added changeset :)
thanks for help with tests!

Rugvip commented on .changeset/few-sloths-raise.md:
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-catalog': minor
+'@backstage/plugin-catalog-react': minor
```
Since this is a non-breaking change and we're on `0.x` we can make these `patch`

dtuite commented:
I released the builds on this PR just now.

alexrybch commented:
rebased and updated tests, please review
